docs/content/docs/reference/nextjs/index.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: 'Introduction'
+---
+
+It is possible to use [`@nhost/react`](/reference/react) in any Next.js page that would be configured to render on the client side.
+
+When rendering a page from the server side, Next.js needs to get some information from the client to determine their authentication status. Such a communication is only available from cookies, and the the Nhost client has been designed to enable such a mechanism.
+
+The cookie is `same-site` and is used only stores the refresh cookie and the expiration date of the JWT so the browser can proactively ask for a new JWT before it expires.
+
+## Signing in
+
+There is no need to activate SSR in the sign-in page as there is not yet any session information to capture. As a result, it is recommended to code it as a classic static page.
+
+The intitial authentication would then work like this diagram:
+
+```mermaid
+sequenceDiagram
+    participant browser as Browser
+    participant next as Next.js
+    participant auth as Hasura Auth
+    browser->>auth: Authenticate
+    activate browser
+    activate auth
+    auth-->>browser: New refresh token + JWT + user data
+    deactivate auth
+    browser->>browser: Set the new refresh token in the cookie
+    deactivate browser
+```
+
+## Refreshing the JWT token
+
+For security considerations, the Nhost JS tools never persist the JWT anywhere in the browser either in `localStorage` or in a cookie, and the JWT expires regularily (15 minutes by default). As a result, the Nhost client regularly call the `hasura-auth` service to get a new JWT from a valid refresh tooken. The new JWT is emitted together with a new refresh token, while the old one is invalidated.
```

Comment:
I found some typos here:

"expires regularily" -> "expires regularly"
"As a result, the Nhost client regularly call" -> "As a result, the Nhost client regularly calls"
"refresh tooken" -> "refresh token"